### PR TITLE
adding token classification

### PR DESCRIPTION
**Pull request by liususan091219**
Adding token classification.

### CODE REVIEW

File: flaml/automl.py [hunk 1]
```diff
@@ -959,7 +970,7 @@ def _prepare_data(self, eval_method, split_ratio, n_splits):
             and self._auto_augment
             and self._state.fit_kwargs.get("sample_weight") is None
             and self._split_type in ["stratified", "uniform"]
-        ):
+        ) and self._state.task != TOKENCLASSIFICATION:
```
Comment (sonichi) [inline flaml/automl.py:973]: move the condition into the ()

File: flaml/data.py [hunk 2]
```diff
@@ -354,12 +357,11 @@ def fit_transform(self, X: Union[DataFrame, np.array], y, task):
                 datetime_columns,
             )
             self._drop = drop
-
         if (
             task in CLASSIFICATION
             or not pd.api.types.is_numeric_dtype(y)
             and task not in NLG_TASKS
-        ):
+        ) and task != TOKENCLASSIFICATION:
```
Comment (sonichi) [inline flaml/data.py:364]: move into ()

File: flaml/ml.py [hunk 3]
```diff
@@ -164,11 +164,25 @@ def metric_loss_score(
...
+                        [len(y_true[x]) < len(y_predict[x]) for x in range(len(y_true))]
+                    ):
+                        y_true_new = np.array(
+                            [[-100] * y_predict.shape[1]] * y_predict.shape[0]
+                        )
+                        for y_true_idx in range(len(y_true_new)):
+                            y_true_new[y_true_idx, : len(y_true[y_true_idx])] = y_true[
+                                y_true_idx
+                            ]
+                        y_true = y_true_new
```
Comment (sonichi) [inline flaml/ml.py:178]: inefficient.
is there a numpy operator to do this?

File: flaml/ml.py [hunk 4]
```diff
@@ -271,6 +286,14 @@ def sklearn_metric_loss_score(
...
             y_true, y_predict, sample_weight=sample_weight
         )
+
+    elif metric_name == "seqeval":
+        from seqeval.metrics import accuracy_score as tok_classification_accuracy_score
+
+        score = tok_classification_accuracy_score(
+            y_true, y_predict, sample_weight=sample_weight
+        )
+
```
Comment (sonichi) [inline flaml/ml.py:296]: why in sklearn_metric_loss_score?

File: flaml/nlp/utils.py [hunk 5]
```diff
@@ -7,28 +7,25 @@
...
-    NLG_TASKS,
     MULTICHOICECLASSIFICATION,
+    TOKENCLASSIFICATION,
+    NLG_TASKS,
 )
 
 
 def load_default_huggingface_metric_for_task(task):
+    from ..data import SEQCLASSIFICATION, SEQREGRESSION, TOKENCLASSIFICATION, NLG_TASKS
+
```
Comment (sonichi) [inline flaml/nlp/utils.py:18]: remove

File: flaml/nlp/utils.py [hunk 6]
```diff
@@ -413,13 +517,20 @@ class HPOArgs:
...
+            A string, the path of the language model file, either a path from huggingface
+            model card huggingface.co/models, or a local path for the model
+        fp16 (:obj:`bool`, `optional`, defaults to :obj:`False`):
+            A bool, whether to use FP16
+        max_seq_length (:obj:`int`, `optional`, defaults to :obj:`128`):
+            An integer, the max length of the sequence
+        pad_to_max_length (:obj:`bool`, `optional`, defaults to :obj:`False`):
+            A bool, whether to pad all samples to model maximum sentence length
+        ckpt_per_epoch (:obj:`int`, `optional`, defaults to :obj:`1`):
+            An integer, the number of checkpoints per epoch
```
Comment (sonichi) [inline flaml/nlp/utils.py:532]: recover

Comment (liususan091219) [reply flaml/nlp/utils.py:532]: recover what

File: test/nlp/test_autohf_tokenclassification.py [hunk 7]
```diff
@@ -720,13 +720,13 @@ def test_tokenclassification():
     automl_settings = {
         "gpu_per_trial": 0,
         "max_iter": 2,
-        "time_budget": 5,
+        "time_budget": 500,
```
Comment (sonichi) [inline test/nlp/test_autohf_tokenclassification.py:723]: too long

File: flaml/nlp/utils.py (hunk 6 above)
Comment (sonichi) [reply flaml/nlp/utils.py:531]: undo the change.

**Review by sonichi — APPROVED** [review]

**Pull request merged**